acceptance/tests/provider/package/gem.rb

```diff
@@ -0,0 +1,96 @@
+test_name "gem provider should install and uninstall" do
+  confine :to, :template => /centos-7-x86_64|windows-2012r2-64/
+  tag 'audit:low'
+
+  require 'puppet/acceptance/common_utils'
+  extend Puppet::Acceptance::PackageUtils
+  extend Puppet::Acceptance::ManifestUtils
+
+  package = 'colorize'
+
+  # https://github.com/puppetlabs/puppet-agent/blob/master/resources/files/puppet-agent.sh
+  puppet_agent_sh_path = '/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin:/opt/puppetlabs/bin'
+
+  wins_ruby_version = '2.4.6-1-x64'
+  wins_ruby_installer_url = "https://github.com/oneclick/rubyinstaller2/releases/download/RubyInstaller-#{wins_ruby_version}/rubyinstaller-#{wins_ruby_version}.exe"
```

Comment:
I have concerns about adding a dependency like this during beaker testing. Could we instead create a new directory that is a symlink to puppet's ruby, and add that to the PATH, while removing the normal one?

Reply:
Is the dependency concern just on Windows? I'm worried about modifying the PATH resulting in an artificial test environment ... if only PUP-6134 were resolved :)